src/index.tsx

```diff
@@ -405,7 +405,9 @@ export function generateTrigger(
       const root = this.getRootDomNode();
       const popupNode = this.getPopupDomNode();
       if (
-        !contains(root, target) &&
+        // mousedown on the target should also close popup when action is contextMenu.
+        // https://github.com/ant-design/ant-design/issues/29853
+        (!contains(root, target) || this.isContextMenuOnly()) &&
```

Comment:
Could help add a test case for this? thx~

Reply:
> Could help add a test case for this? thx~

I'm not familiar with enzyme
Just tried adding it in the end of basic.test.js -> 'contextMenu works' 

tried both `wrapper.trigger('mouseDown');` and  `wrapper.find('.target').simulate('mouseDown');`
neither close the popup for me.  

However, it does work in the storybook, So I'm not quite sure what to do.

Reply:
You need simulate `contextMenu` first. And then simulate `mouseDown` next to check if it closed.

Reply:
> You need simulate `contextMenu` first. And then simulate `mouseDown` next to check if it closed.

This is what I did, I tried both way to mouseDown but neither closes the popup
```
    it('contextMenu works', () => {
      const wrapper = mount(
        <Trigger
          action={['contextMenu']}
          popupAlign={placementAlignMap.left}
          popup={<strong>trigger</strong>}
        >
          <div className="target">contextMenu</div>
        </Trigger>,
      );

      wrapper.trigger('contextMenu');

      expect(wrapper.isHidden()).toBeFalsy();

      wrapper.trigger('mouseDown');
      wrapper.find('.target').simulate('mouseDown');
      expect(wrapper.isHidden()).toBeTruthy();
    });
```

Reply:
@zombieJ simulate mouseDown on document is pretty tricky, this is the easiest work around I can find, please take a look

Reply:
> You need simulate `contextMenu` first. And then simulate `mouseDown` next to check if it closed.

done